test/test_torch.py

```diff
@@ -4600,6 +4570,36 @@ def test_masked_select_discontiguous(self, device):
                 torch.masked_select(v, m, out=out_dc)
                 self.assertEqual(out_dc, expected, atol=0, rtol=0)
 
+    @dtypes(*torch.testing.get_all_dtypes())
+    def test_masked_fill(self, device, dtype):
+        dt = dtype
+        with warnings.catch_warnings(record=True) as w:
+            warnings.simplefilter("always")
+            for dtype in [torch.uint8, torch.bool]:
+                num_dest = 10
+                dst = torch.tensor([0, 0, 0, 0, 0, 0, 0, 0, 0, 0], dtype=dt)
+                mask = torch.rand(num_dest).mul(2).floor().to(dtype)
```

Comment:
this is a complicated way of saying `torch.randint(2, (num_dest,), dtype=dtype[1])` (assuming your mask dtype is `dtype[1]`